lib/transport.dart

```diff
@@ -210,6 +210,11 @@ abstract class TransportStream {
   /// A sink for writing data and/or headers to the remote end.
   StreamSink<StreamMessage> get outgoingMessages;
 
+  /// Set the cancellation handler on this stream.
+  ///
+  /// The handler will be called if the stream receives an RST_STREAM frame.
+  void onCancel(void value(int errorCode));
```

Comment:
The dual for the `set terminatedHandler` would then be a `void terminate()` method which will send the rst frame.

Would it then make sense to no longer add errors to the `incomingMessages` stream and only report the termination via the handler? Or do you think it's better to have both?